components/eth-price-oracle/src/lib.rs

```diff
@@ -1,159 +1,127 @@
-mod trigger;
-use trigger::{decode_trigger_event, encode_trigger_output, Destination};
-use wavs_wasi_chain::http::{fetch_json, http_request_get};
-pub mod bindings;
+use crate::bindings::wavs::worker::layer_types::{TriggerData, TriggerDataEthContractEvent};
 use crate::bindings::{export, Guest, TriggerAction};
+use alloy_sol_types::SolValue;
+use anyhow::Result;
 use serde::{Deserialize, Serialize};
-use wstd::{http::HeaderValue, runtime::block_on};
+use wavs_wasi_chain::decode_event_log_data;
+use wavs_wasi_chain::http::{fetch_json, http_request_get};
+use wstd::runtime::block_on;
+
+pub mod bindings;
+
+/// Represents the destination where the trigger output should be sent
+pub enum Destination {
+    Ethereum,
+    CliOutput,
+}
+
+/// Decodes incoming trigger event data into its components
+fn decode_trigger_event(trigger_data: TriggerData) -> Result<(u64, Vec<u8>, Destination)> {
+    match trigger_data {
+        TriggerData::EthContractEvent(TriggerDataEthContractEvent { log, .. }) => {
+            let event: solidity::NewTrigger = decode_event_log_data!(log)?;
+            let trigger_info = solidity::TriggerInfo::abi_decode(&event._triggerInfo, false)?;
+            Ok((trigger_info.triggerId, trigger_info.data.to_vec(), Destination::Ethereum))
+        }
+        TriggerData::Raw(data) => Ok((0, data.clone(), Destination::CliOutput)),
+        _ => Err(anyhow::anyhow!("Unsupported trigger data type")),
+    }
+}
+
+/// Encodes the output data for submission back to Ethereum
+fn encode_trigger_output(trigger_id: u64, output: impl AsRef<[u8]>) -> Vec<u8> {
+    solidity::DataWithId { triggerId: trigger_id, data: output.as_ref().to_vec().into() }
+        .abi_encode()
+}
+
+mod solidity {
+    use alloy_sol_macro::sol;
+    pub use ITypes::*;
+    sol!("../../src/interfaces/ITypes.sol");
+}
```

Comment:
Does this really need to be a module?

Reply:
does not have to be, done here like this just for simplicity

mentioned elsewhere that the new output would be

```
sol!("../../src/interfaces/ITypes.sol");
ITypes::ContractType
```

I figured this was nicer DevEx so that solidity:: is always the namespace, rather than the arb filename